diff -Naur pgPoint-0.0.1/pgPoint.py pgPoint-0.0.1-gentoo/pgPoint.py
--- pgPoint-0.0.1/pgPoint.py	2003-12-02 01:02:42.000000000 +0900
+++ pgPoint-0.0.1-gentoo/pgPoint.py	2004-05-08 23:16:49.029211447 +0900
@@ -1,4 +1,4 @@
-# -*- coding: japanese.euc-jp -*-
+# -*- coding: euc-jp -*-
 
 # for Python 2.2
 # それ以前だとげねれーたがないから動かないけどまいっか
@@ -15,15 +15,10 @@
 import pygame.font
 from pygame.locals import *
 
-# ここをお手元のフォントファイル名に書き換えよう(なげやり)
-_font_filename = "aquafont.ttf"
-#_font_filename = r"C:\WINNT\Fonts\msgothic.ttc" # Windowsの場合こんな感じかも
-#_font_filename = "/usr/share/fonts/truetype/japanese/kochi-gothic.ttf" # UNIX / Linuxの場合こんな感じかも
-
 _version = "0.0.1"
 _screen_size = (640,480)
 _screen_surface = None
-_encoding = "japanese.euc-jp"
+_encoding = "euc-jp"
 _background_color = (0,0,63)
 
 def Init():
@@ -201,9 +196,9 @@
 		self._position = (self._position[0],_screen_size[1] - rect[3] - val)
 
 class String(Item):
-	def __init__(self,string,position,size,color):
+	def __init__(self,string,position,size,color,fontname="/usr/X11R6/lib/X11/fonts/truetype/kochi-gothic.ttf"):
 		Item.__init__(self)
-		font = pygame.font.Font(_font_filename,size)
+		font = pygame.font.Font(fontname,size)
 		self._surface = font.render(unicode(string,_encoding),1,color,_background_color).convert()
 		self._position = position
 
diff -Naur pgPoint-0.0.1/sample.py pgPoint-0.0.1-gentoo/sample.py
--- pgPoint-0.0.1/sample.py	2003-12-02 01:02:42.000000000 +0900
+++ pgPoint-0.0.1-gentoo/sample.py	2004-05-08 23:20:40.654287494 +0900
@@ -1,4 +1,4 @@
-# -*- coding: japanese.euc-jp -*-
+# -*- coding: euc-jp -*-
 
 import random
 import pgPoint
