*** ../../drbd-0.6.13/scripts/Makefile	Thu Apr  1 16:35:14 2004
--- Makefile	Tue Mar 22 23:21:16 2005
***************
*** 42,47 ****
--- 42,51 ----
    INITD=etc/rc.d/init.d/
  endif
  
+ ifeq ($(DIST),gentoo)
+   INITD=$(PREFIX)sbin/
+ endif
+ 
  ifeq ($(INITD),)
     INITD=etc/init.d/
     $(warning Can not guess distribution INITD, please fix Makefile.vars)
***************
*** 54,68 ****
  install:
  	mkdir -p $(PREFIX)etc/ha.d/resource.d
  	install -D -m 755 drbd $(PREFIX)$(INITD)drbd
! 	@ if [ ! -e $(PREFIX)etc/drbd.conf ]; then \
! 		install -D -m 644 drbd.conf $(PREFIX)etc/drbd.conf; \
! 	fi
  	ln -sf ../../../$(INITD)drbd $(PREFIX)etc/ha.d/resource.d/datadisk
! ifeq ($(DIST),suse)
! 	ln -sf ../$(INITD)drbd $(PREFIX)sbin/rcdrbd
! endif
! 	@ echo
! 	@ echo "Don't forget to run update-rc.d or chkconfig"
  
  clean:
  	rm -f *~
--- 58,72 ----
  install:
  	mkdir -p $(PREFIX)etc/ha.d/resource.d
  	install -D -m 755 drbd $(PREFIX)$(INITD)drbd
! #	@ if [ ! -e $(PREFIX)etc/drbd.conf ]; then \
! #		install -D -m 644 drbd.conf $(PREFIX)etc/drbd.conf; \
! #	fi
  	ln -sf ../../../$(INITD)drbd $(PREFIX)etc/ha.d/resource.d/datadisk
! #ifeq ($(DIST),suse)
! #	ln -sf ../$(INITD)drbd $(PREFIX)sbin/rcdrbd
! #endif
! #	@ echo
! #	@ echo "Don't forget to run update-rc.d or chkconfig"
  
  clean:
  	rm -f *~
