*** ../../drbd-0.6.13/drbd/Makefile	Tue Jul 13 16:03:56 2004
--- Makefile	Tue Mar 22 23:17:29 2005
***************
*** 134,145 ****
  	test -n "$(KERNELRELEASE)"
  	install -D -m 644 drbd.o "$(PREFIX)lib/modules/$(KERNELRELEASE)/$(MODSUBDIR)/drbd.o"
  	install -d -m 700 "$(PREFIX)var/lib/drbd"
!   ifeq ($(shell uname -r),$(KERNELRELEASE))
! 	/sbin/depmod -a || /sbin/depmod -e drbd.o 2>&1 >/dev/null || true
!   else
! 	[ -e $(LINUX)/System.map ] && \
! 	   /sbin/depmod -F $(LINUX)/System.map -e ./drbd.o 2>&1 >/dev/null || true
!   endif
  
    depmod: drbd.o
  	[ -e $(LINUX)/System.map ] && \
--- 134,145 ----
  	test -n "$(KERNELRELEASE)"
  	install -D -m 644 drbd.o "$(PREFIX)lib/modules/$(KERNELRELEASE)/$(MODSUBDIR)/drbd.o"
  	install -d -m 700 "$(PREFIX)var/lib/drbd"
! #  ifeq ($(shell uname -r),$(KERNELRELEASE))
! #	/sbin/depmod -a || /sbin/depmod -e drbd.o 2>&1 >/dev/null || true
! #  else
! #	[ -e $(LINUX)/System.map ] && \
! #	   /sbin/depmod -F $(LINUX)/System.map -e ./drbd.o 2>&1 >/dev/null || true
! #  endif
  
    depmod: drbd.o
  	[ -e $(LINUX)/System.map ] && \
