--- cvs-1.11.13/src/cvs.h.orig	2004-04-03 19:56:37.000000000 +0900
+++ cvs-1.11.13/src/cvs.h	2004-04-03 19:57:00.000000000 +0900
@@ -61,6 +61,10 @@
 #define O_NONBLOCK 0x0004
 #endif
 
+#ifdef HAVE_FCNTL_H
+#include <fcntl.h>
+#endif
+
 #ifndef F_SETFL
 extern int fcntl (int, int, int);
 #define F_GETFL 3
--- cvs-1.11.13/src/tsocks.c.orig	2004-04-03 19:56:37.000000000 +0900
+++ cvs-1.11.13/src/tsocks.c	2004-04-03 20:15:53.000000000 +0900
@@ -833,6 +833,7 @@
 	    pkt[2+nmethods++] = method;
 	    debug("available auth method[%i] = %s (0x%02x)", 
 		  i, socks5_getauthname(method), method);
+	    break;
 	}
     }
     pkt[1] = nmethods;
