--- qt-x11-free-3.3.4.orig/src/kernel/qapplication_x11.cpp	2005-05-28 23:53:42.000000000 +0900
+++ qt-x11-free-3.3.4/src/kernel/qapplication_x11.cpp	2005-05-28 23:58:00.000000000 +0900
@@ -3227,7 +3227,7 @@
 		if ( inPopupMode() ) // no focus widget, see if we have a popup
 		    keywidget = (QETWidget*) activePopupWidget();
 		else if ( widget )
-		    keywidget = (QETWidget*)widget->topLevelWidget();
+			keywidget = widget->focusWidget()?(QETWidget*)widget->focusWidget():widget;
 	    }
 	}
     }
@@ -3258,7 +3258,7 @@
 		    if ( inPopupMode() ) // no focus widget, see if we have a popup
 		        keywidget = (QETWidget*) activePopupWidget();
 		    else if ( widget )
-		        keywidget = (QETWidget*)widget->topLevelWidget();
+				keywidget = widget->focusWidget()?(QETWidget*)widget->focusWidget():widget;
 	        }
 	    }
         }
