--- Makefile.in.orig	2004-12-19 17:51:34.000000000 +0900
+++ Makefile.in	2004-12-19 14:40:15.000000000 +0900
@@ -765,7 +765,8 @@
 	all-target-zlib \
 	all-target-boehm-gc \
 	all-target-qthreads \
-	all-target-rda
+	all-target-rda \
+	all-target-phobos
 
 # This is a list of the configure targets for all of the modules which
 # are compiled using the target tools.
@@ -785,7 +786,8 @@
 	configure-target-zlib \
 	configure-target-boehm-gc \
 	configure-target-qthreads \
-	configure-target-rda
+	configure-target-rda \
+	configure-target-phobos
 
 # This is a list of the check targets for all of the modules which are
 # compiled using $(TARGET_FLAGS_TO_PASS).
@@ -802,7 +804,8 @@
 	check-target-zlib \
 	check-target-boehm-gc \
 	check-target-qthreads \
-	check-target-rda
+	check-target-rda \
+	check-target-phobos
 
 # This is a list of the install targets for all of the modules which are
 # compiled using $(TARGET_FLAGS_TO_PASS).
@@ -821,7 +824,8 @@
 	install-target-zlib \
 	install-target-boehm-gc \
 	install-target-qthreads \
-	install-target-rda
+	install-target-rda \
+	install-target-phobos
 
 # This is a list of the targets for which we can do a clean-{target}.
 CLEAN_MODULES = \
@@ -902,7 +906,8 @@
 	clean-target-zlib \
 	clean-target-boehm-gc \
 	clean-target-qthreads \
-	clean-target-rda
+	clean-target-rda \
+	clean-target-phobos
 
 # All of the x11 modules that can be cleaned
 CLEAN_X11_MODULES = \
@@ -1734,6 +1739,8 @@
 all-tix: all-tcl all-tk
 all-wdiff:
 configure-target-rda: $(ALL_GCC_C)
+configure-target-phobos: $(ALL_GCC_C)
+all-target-phobos: all-target-libiberty
 configure-target-winsup: $(ALL_GCC_C)
 all-target-winsup: all-target-libiberty all-target-libtermcap
 all-uudecode: all-libiberty
@@ -1750,6 +1757,7 @@
 install-gdb: install-tcl install-tk install-itcl install-tix install-libgui
 install-sid: install-tcl install-tk
 
+
 # We put install-opcodes before install-binutils because the installed
 # binutils might be on PATH, and they might need the shared opcodes
 # library.
@@ -1777,6 +1785,7 @@
 all-target-boehm-gc: configure-target-boehm-gc
 all-target-qthreads: configure-target-qthreads
 all-target-rda: configure-target-rda
+all-target-phobos: configure-target-phobos
 
 
 ### other supporting targets
