diff -Naur phobos.orig/Makefile.in phobos/Makefile.in
--- phobos.orig/Makefile.in	2004-12-19 21:32:17.000000000 +0900
+++ phobos/Makefile.in	2004-12-19 23:11:43.000000000 +0900
@@ -18,7 +18,7 @@
 AR = @AR@
 RANLIB = @RANLIB@
 
-CFLAGS=@CFLAGS@ @DEFS@ -I etc/c/stlsoft -I .
+MYCPPFLAGS=@CFLAGS@ @DEFS@ -I etc/c/stlsoft -I .
 
 # Because parts of Phobos are generated (and are in flux), we need
 # to prevent this build from getting tripped up on an already installed
@@ -31,7 +31,7 @@
 DMD=@DMD@
 
 # For recls
-CXXFLAGS=@CXXFLAGS@ @DEFS@ -I $(srcdir)/etc/c/stlsoft -fno-access-control -fno-exceptions
+CPPFLAGS=@CXXFLAGS@ @DEFS@ -I $(srcdir)/etc/c/stlsoft -fno-access-control -fno-exceptions
 
 srcdir=@srcdir@
 VPATH = @srcdir@
@@ -44,7 +44,7 @@
 all: libphobos.a
 
 %.o : %.c
-	$(CC) -o $@ $(CFLAGS) -c $<
+	$(CC) -o $@ $(CFLAGS) $(MYCPPFLAGS) -c $<
 
 %.o : %.d
 	$(DMD) -o $@ $(DFLAGS) -I $(srcdir) -I $(srcdir)/internal/gc -c $<
@@ -167,18 +167,18 @@
 	$(RANLIB) $@
 
 install: gcc/config.d libphobos.a
-	rm -r -f $(includedir)/d
-	mkdir -p $(includedir)/d
-	cp -p -R $(srcdir)/*.d $(includedir)/d
-	cp -p -R $(srcdir)/etc $(includedir)/d
-	cp -p -R $(srcdir)/gcc $(includedir)/d
-	cp -p -R $(srcdir)/internal $(includedir)/d
-	cp -p -R $(srcdir)/std $(includedir)/d
-	cp -p gcc/config.d $(includedir)/d/gcc
-	if test -f gcc/configunix.d; then cp -p gcc/configunix.d $(includedir)/d/gcc; fi
-	cp -p phobos-ver-syms $(includedir)/d
-	cp -p libphobos.a $(libdir)
-	cp -p libphobos.spec $(libdir)
+	rm -r -f $(DESTDIR)$(includedir)/d
+	mkdir -p $(DESTDIR)$(includedir)/d
+	cp -p -R $(srcdir)/*.d $(DESTDIR)$(includedir)/d
+	cp -p -R $(srcdir)/etc $(DESTDIR)$(includedir)/d
+	cp -p -R $(srcdir)/gcc $(DESTDIR)$(includedir)/d
+	cp -p -R $(srcdir)/internal $(DESTDIR)$(includedir)/d
+	cp -p -R $(srcdir)/std $(DESTDIR)$(includedir)/d
+	cp -p gcc/config.d $(DESTDIR)$(includedir)/d/gcc
+	if test -f gcc/configunix.d; then cp -p gcc/configunix.d $(DESTDIR)$(includedir)/d/gcc; fi
+	cp -p phobos-ver-syms $(DESTDIR)$(includedir)/d
+	cp -p libphobos.a $(DESTDIR)$(LIBPATH)
+	cp -p libphobos.spec $(DESTDIR)$(LIBPATH)
 
 clean:
 	echo 'Removing files...'
diff -Naur phobos.orig/configure phobos/configure
--- phobos.orig/configure	2004-12-19 21:32:17.000000000 +0900
+++ phobos/configure	2004-12-19 21:35:45.000000000 +0900
@@ -2825,6 +2825,7 @@
   if $as_executable_p "$as_dir/$ac_word$ac_exec_ext"; then
     ac_cv_prog_d_have_dmd="yes"
     echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
+    DMD="`which gdc`"
     break 2
   fi
 done
