diff -ruN im-sdk-trunk-svn2172.orig/iiimxcf/htt_xbe/lib/IMdkit/r6im/R6IMProtocol.cpp im-sdk-trunk-svn2172/iiimxcf/htt_xbe/lib/IMdkit/r6im/R6IMProtocol.cpp
--- im-sdk-trunk-svn2172.orig/iiimxcf/htt_xbe/lib/IMdkit/r6im/R6IMProtocol.cpp	2005-01-31 15:31:49.000000000 +0900
+++ im-sdk-trunk-svn2172/iiimxcf/htt_xbe/lib/IMdkit/r6im/R6IMProtocol.cpp	2005-01-31 16:20:09.565867067 +0900
@@ -397,7 +397,7 @@
       }
       else if (!strcmp(icp->name, XNBackgroundPixmap)) {
 	icp->value = new Pixmap;
-	ic->get_preedit_bg_pixmap(*(Pixmap*)icp->value);
+	ic->get_preedit_bg_pixmap(*(CARD32*)(Pixmap*)icp->value);
 	icp->value_length = sizeof(Pixmap);
       }
       else if (!strcmp(icp->name, XNLineSpace)) {
@@ -458,7 +458,7 @@
       }
       else if (!strcmp(icp->name, XNBackgroundPixmap)) {
 	icp->value = new Pixmap;
-	ic->get_status_bg_pixmap(*(Pixmap*)icp->value);
+	ic->get_status_bg_pixmap(*(CARD32*)(Pixmap*)icp->value);
 	icp->value_length = sizeof(Pixmap);
       }
       else if (!strcmp(icp->name, XNLineSpace)) {
diff -ruN im-sdk-trunk-svn2172.orig/iiimxcf/htt_xbe/lib/IMdkit/ximp/XIMPInputContext.cpp im-sdk-trunk-svn2172/iiimxcf/htt_xbe/lib/IMdkit/ximp/XIMPInputContext.cpp
--- im-sdk-trunk-svn2172.orig/iiimxcf/htt_xbe/lib/IMdkit/ximp/XIMPInputContext.cpp	2005-01-31 15:31:49.000000000 +0900
+++ im-sdk-trunk-svn2172/iiimxcf/htt_xbe/lib/IMdkit/ximp/XIMPInputContext.cpp	2005-01-31 16:21:01.475191555 +0900
@@ -1438,7 +1438,7 @@
     Atom atom1, atom2, atom3;
     CARD32 l[3];
     char *ctext;
-    CARD32 *feedback;
+    unsigned long *feedback;
     int n;
 
     l[0] = draw->caret; l[1] = draw->chg_first; l[2] = draw->chg_length;
@@ -1542,7 +1542,7 @@
   int input_context_id = ximp_protocol->getICID(this);
   Atom atom1, atom2;
   char *ctext;
-  CARD32 *feedback;
+  unsigned long *feedback;
   int len;
   int n;
 
