diff -ruN iiimf-trunk-svn2421.orig/iiimxcf/htt_xbe/lib/IMdkit/r6im/XInputContext.cpp iiimf-trunk-svn2421/iiimxcf/htt_xbe/lib/IMdkit/r6im/XInputContext.cpp
--- iiimf-trunk-svn2421.orig/iiimxcf/htt_xbe/lib/IMdkit/r6im/XInputContext.cpp	2005-04-01 11:48:42.000000000 +0900
+++ iiimf-trunk-svn2421/iiimxcf/htt_xbe/lib/IMdkit/r6im/XInputContext.cpp	2005-04-01 21:30:30.000000000 +0900
@@ -165,10 +165,10 @@
 			     XNPreeditDrawCallback, preedit_draw_cb,
 			     XNPreeditCaretCallback, preedit_caret_cb,
 			     XNPreeditDoneCallback, preedit_done_cb,
-			     0);
+			     NULL);
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
     XFree(preedit_attr);
   }
   return;
@@ -182,10 +182,10 @@
 			    XNStatusStartCallback, status_start_cb,
 			    XNStatusDrawCallback, status_draw_cb,
 			    XNStatusDoneCallback, status_done_cb,
-			    0);
+			    NULL);
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
     XFree(status_attr);
   }
 }
@@ -199,7 +199,7 @@
 		 // private XIC attribute
 		 "commitStringCallback", commit_string_cb,
 		 "forwardEventCallback", forward_event_cb,
-		 0);
+		 NULL);
   return;
 }
 
@@ -216,7 +216,7 @@
       // such cases, we allow to set client_win again.
       XSetICValues(xic,
 		   XNClientWindow, client_win,
-		   0);
+		   NULL);
       filters &= ~FLT_DESTROY;
       select_destroy_clientwin_emask();
       register_clientwin_destroy_filter();
@@ -273,7 +273,7 @@
     if (xic)
       XSetICValues(xic,
 		   XNFocusWindow, focus_win,
-		   0);
+		   NULL);
   }
   return;
 }
@@ -361,7 +361,7 @@
   if (reply_back && xic) {
     XSetICValues(xic,
 		 XNPreeditState, XIMPreeditEnable,
-		 0);
+		 NULL);
   }
   is_conv_on = 1;
 }
@@ -371,7 +371,7 @@
   if (reply_back && xic) {
     XSetICValues(xic,
 		 XNPreeditState, XIMPreeditDisable,
-		 0);
+		 NULL);
   }
   is_conv_on = 0;
 }
@@ -393,7 +393,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
   return;
 }
@@ -417,7 +417,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
   return;
 }
@@ -439,7 +439,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
   if (width == 0 || height == 0) {
     // do geometry negotiation
@@ -450,7 +450,7 @@
     if (xic)
       XGetICValues(xic,
 		   XNPreeditAttributes, preedit_attr,
-		   0);
+		   NULL);
     XFree(preedit_attr);
     if (area2) {
       if (width == 0) {
@@ -491,7 +491,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
   return;
 }
@@ -509,7 +509,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
 }
 void
@@ -526,7 +526,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
 }
 void
@@ -543,7 +543,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
 }
 void
@@ -560,7 +560,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
 }
 void
@@ -576,7 +576,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
 }
 void
@@ -593,7 +593,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNPreeditAttributes, preedit_attr,
-		 0);
+		 NULL);
   XFree(preedit_attr);
 }
 
@@ -617,7 +617,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
   return;
 }
@@ -639,7 +639,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
   if (width == 0 || height == 0) {
     // do geometry negotiation
@@ -650,7 +650,7 @@
     if (xic)
       XGetICValues(xic,
 		   XNStatusAttributes, status_attr,
-		   0);
+		   NULL);
     XFree(status_attr);
     if (area2) {
       if (width == 0 ||
@@ -693,7 +693,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
   return;
 }
@@ -711,7 +711,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
 }
 void
@@ -728,7 +728,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
 }
 void
@@ -745,7 +745,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
 }
 void
@@ -762,7 +762,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
 }
 void
@@ -779,7 +779,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
 }
 void
@@ -796,7 +796,7 @@
   if (xic)
     XSetICValues(xic,
 		 XNStatusAttributes, status_attr,
-		 0);
+		 NULL);
   XFree(status_attr);
 }
 
