diff -ruN iiimf-r12_2-svn2578.orig/iiimsf/src/IIIMP_IMState.cpp iiimf-r12_2-svn2578/iiimsf/src/IIIMP_IMState.cpp
--- iiimf-r12_2-svn2578.orig/iiimsf/src/IIIMP_IMState.cpp	2005-04-28 21:37:51.000000000 +0900
+++ iiimf-r12_2-svn2578/iiimsf/src/IIIMP_IMState.cpp	2005-04-28 21:37:33.000000000 +0900
@@ -756,14 +756,27 @@
 	      HotKeyList *hlist = NULL;
 	      string conffile = ".iiim/le.xml.conf";
 
-#if 0 /* disable reading the user-specific configuration ATM */
-	      /* load the user-specific configuration */
+	      /* load the user-specific configuration via the namespace I/O */
 	      uconf = iiim_le_xmlconf_new(conffile.c_str());
 	      if (uconf) {
 		  iiim_le_xmlconf_load_with_nsio(uconf, idt);
 		  hlist = iiim_le_xmlconf_get_hotkey_list(uconf, curr_input_lang.get_charstr());
 	      }
-#endif
+	      if (hlist == NULL) {
+		  /* try to read it from the local machine directly */
+		  struct passwd *pass = getpwnam(idt->user_name);
+
+		  if (pass != NULL) {
+		      conffile = string(pass->pw_dir) + string("/.iiim/le.xml.conf");
+		      if (uconf)
+			  iiim_le_xmlconf_free(uconf);
+		      uconf = iiim_le_xmlconf_new(conffile.c_str());
+		      if (uconf) {
+			  iiim_le_xmlconf_load_with_nsio(uconf, idt);
+			  hlist = iiim_le_xmlconf_get_hotkey_list(uconf, curr_input_lang.get_charstr());
+		      }
+		  }
+	      }
 	      if (hlist == NULL) {
 		  /* try to read the global configuration */
 		  if (uconf)
