--- iiimf-r12_2-svn2578/iiimgcf/gtkimcontextiiim.c~	2005-04-27 17:24:23.000000000 +0900
+++ iiimf-r12_2-svn2578/iiimgcf/gtkimcontextiiim.c	2005-04-28 09:33:36.000000000 +0900
@@ -568,8 +568,8 @@
 	  DEBUG_DO (g_message ("ui_status_change"));
 	  status_callback (context_iiim);
 	  break;
-	case IIIMCF_EVENT_TYPE_UI_STATUS_END:
-	  DEBUG_DO (g_message ("ui_status_end"));
+	case IIIMCF_EVENT_TYPE_UI_STATUS_DONE:
+	  DEBUG_DO (g_message ("ui_status_done"));
 	  status_callback (context_iiim);
 	  break;
 	case IIIMCF_EVENT_TYPE_AUX_START:
