diff -uNr im-sdk-src-r12_1.orig/iiimgcf/gtkimcontextiiim.h im-sdk-src-r12_1/iiimgcf/gtkimcontextiiim.h
--- im-sdk-src-r12_1.orig/iiimgcf/gtkimcontextiiim.h	2004-11-03 12:28:52.487732624 +1000
+++ im-sdk-src-r12_1/iiimgcf/gtkimcontextiiim.h	2004-11-03 12:29:22.308199224 +1000
@@ -119,6 +119,13 @@
 				       IIIMCF_event ev);
 extern void im_context_aux_get_values (GtkIMContextIIIM *context_iiim,
 				       IIIMCF_event ev);
+extern void im_context_switcher_set_language_engine_list (GtkIMContextIIIM *context_iiim,
+                                              gchar *le_list);
+
+extern void _XRegisterFilterByType();
+
+extern void _XUnregisterFilter();
+
 G_END_DECLS
 
 #endif
diff -uNr im-sdk-src-r12_1.orig/iiimgcf/imaux.c im-sdk-src-r12_1/iiimgcf/imaux.c
--- im-sdk-src-r12_1.orig/iiimgcf/imaux.c	2004-11-03 12:28:52.488732472 +1000
+++ im-sdk-src-r12_1/iiimgcf/imaux.c	2004-11-03 12:29:23.205062880 +1000
@@ -234,8 +234,8 @@
 iiim_setup_aux_object (IIIMCF_handle iiim)
 {
   IIIMF_status st;
-  IIIMCF_object_descriptor *pdesc;
-  IIIMCF_object_descriptor **ppdescs;
+  const IIIMCF_object_descriptor *pdesc;
+  const IIIMCF_object_descriptor **ppdescs;
   IIIMCF_downloaded_object *pobjs;
   int i, n1, n2;
 
@@ -247,7 +247,7 @@
     return st;
 
   pobjs = g_new0 (IIIMCF_downloaded_object, n1);
-  ppdescs = g_new0 (IIIMCF_object_descriptor*, n1);
+  ppdescs = g_new0 (const IIIMCF_object_descriptor *, n1);
 
   for (i = 0, n2 = 0; i < n1; i++, pdesc++)
     {
@@ -881,8 +881,8 @@
     int i, n;
     int aux_data_t_n;
     int aux_name_len, aux_name_n;
-    int integer_list_n, string_list_n;
-    int string_n;
+    int integer_list_n = 0, string_list_n = 0;
+    int string_n = 0;
     int *pstring_len;
 
 #define ROUNDUP(n) ((n + sizeof(int) - 1) / sizeof(int) * sizeof(int))
