diff -ruN im-sdk-trunk-svn2172.orig/leif/configure.ac im-sdk-trunk-svn2172/leif/configure.ac
--- im-sdk-trunk-svn2172.orig/leif/configure.ac	2005-01-31 15:31:34.000000000 +0900
+++ im-sdk-trunk-svn2172/leif/configure.ac	2005-01-31 16:50:20.196575057 +0900
@@ -103,6 +103,7 @@
 IM_SYSTEM_SUBST
 IM_X_SUBST
 
+m4_pattern_allow(AM_ET)
 AC_OUTPUT(Makefile \
 	  default/Makefile \
 	  ude/Makefile \
diff -ruN im-sdk-trunk-svn2172.orig/leif/unit/codetable/Makefile.am im-sdk-trunk-svn2172/leif/unit/codetable/Makefile.am
--- im-sdk-trunk-svn2172.orig/leif/unit/codetable/Makefile.am	2005-01-31 15:28:47.000000000 +0900
+++ im-sdk-trunk-svn2172/leif/unit/codetable/Makefile.am	2005-01-31 16:48:41.448938870 +0900
@@ -1,6 +1,6 @@
 bindir = $(IM_LOCALEDIR)/UNIT/common
 
-COMMON_LIB = ../xaux_common/libxaux_common.la
+COMMON_LIB = -L../xaux_common -lxaux_common
 CTIM_LIB = ../ctim/ctim.la
 COMMON_INCLUDES = -I../ctim -I../include
 
diff -ruN im-sdk-trunk-svn2172.orig/leif/unit/ctim/Makefile.am im-sdk-trunk-svn2172/leif/unit/ctim/Makefile.am
--- im-sdk-trunk-svn2172.orig/leif/unit/ctim/Makefile.am	2005-01-31 15:28:50.000000000 +0900
+++ im-sdk-trunk-svn2172/leif/unit/ctim/Makefile.am	2005-01-31 16:49:06.343563846 +0900
@@ -2,5 +2,5 @@
 ctim_la_SOURCES = ctim.c codetable.c ctfilter.c keymap.c
 ctim_la_CFLAGS = $(SHLIBCFLAGS) $(THREAD_CXX_CFLAGS) -I$(IM_INCLUDEDIR)/iml -I../include $(XML_CFLAGS)
 ctim_la_LDFLAGS = -module -rpath $(moduledir) $(SHLIBLDFLAGS) $(THREAD_LDFLAGS) $(XML_LDFLAGS)
-ctim_la_LIBADD = ../xaux_common/libxaux_common.la
+ctim_la_LIBADD = -L../xaux_common -lxaux_common
 module_LTLIBRARIES = ctim.la
diff -ruN im-sdk-trunk-svn2172.orig/leif/unit/leif/Makefile.am im-sdk-trunk-svn2172/leif/unit/leif/Makefile.am
--- im-sdk-trunk-svn2172.orig/leif/unit/leif/Makefile.am	2005-01-31 15:28:46.000000000 +0900
+++ im-sdk-trunk-svn2172/leif/unit/leif/Makefile.am	2005-01-31 16:49:27.889646254 +0900
@@ -4,7 +4,7 @@
 unitle_la_CFLAGS = $(SHLIBCFLAGS) $(THREAD_CXX_CFLAGS) \
                     -I$(IM_INCLUDEDIR)/iml -I../include $(XML_CFLAGS)
 unitle_la_LDFLAGS = -module -rpath $(IM_LEIFDIR) $(SHLIBLDFLAGS) $(THREAD_LDFLAGS) $(XML_LDFLAGS)
-unitle_la_LIBADD = $(THREAD_LIBS) ../xaux_common/libxaux_common.la
+unitle_la_LIBADD = $(THREAD_LIBS) -L../xaux_common -lxaux_common
 unitle_la_SOURCES = define.c unit_input.c le.c hotkey_config.c \
                      key_event.c le_proc.c iml.c encode_draw.c
 noinst_LTLIBRARIES = unitle.la
