diff -ruN im-sdk-trunk-svn2172.orig/iiimxcf/xiiimp.so/iiimp/iiimpDL.c im-sdk-trunk-svn2172/iiimxcf/xiiimp.so/iiimp/iiimpDL.c
--- im-sdk-trunk-svn2172.orig/iiimxcf/xiiimp.so/iiimp/iiimpDL.c	2005-01-31 15:31:44.000000000 +0900
+++ im-sdk-trunk-svn2172/iiimxcf/xiiimp.so/iiimp/iiimpDL.c	2005-01-31 16:24:36.718132505 +0900
@@ -51,6 +51,9 @@
 #include "genutil.h"
 #include "Xlcint.h"
 
+#undef _LP64
+/*undefine it for all Red Hat architectures. Yu Shao*/
+
 #ifdef _LP64
 #if defined(__sparcv9)
 #define	_MACH64_NAME		"sparcv9"
