diff -ur jd-10060106.orig/src/winmain.cpp jd-10060106/src/winmain.cpp
--- jd-10060106.orig/src/winmain.cpp	2006-01-01 01:02:20.000000000 +0900
+++ jd-10060106/src/winmain.cpp	2006-01-13 00:17:50.000000000 +0900
@@ -20,6 +20,9 @@
 {
     setlocale( LC_ALL, "ja_JP.UTF-8" );
 
+    // init glib thread
+    if( !Glib::thread_supported() ) Glib::thread_init();
+
     // 全体設定ロード
     bool init = !( CONFIG::init_config() );
 
