diff -r -u jd-150_060601.org/README jd-150_060601/README
--- jd-150_060601.org/README	2006-05-31 20:41:37.000000000 +0900
+++ jd-150_060601/README	2006-06-05 04:56:52.000000000 +0900
@@ -377,6 +377,13 @@
 //
 
 
+64bit対応パッチを当てた
+articleviewで検索したときに再描画がかからない時があるバグを修正
+articleviewで検索対象の文字列が最後のレスに含まれているとリンク下線がずれるバグを修正
+一度articleviewを閉じないと最初にブックマークしたレスにジャンプ出来ないバグを修正
+ダブルクリックで板やスレを開けないバグを修正
+
+
 - 1.50.060601
 
 誤った正規表現を入力すると落ちるバグを修正
diff -r -u jd-150_060601.org/src/article/articleviewbase.cpp jd-150_060601/src/article/articleviewbase.cpp
--- jd-150_060601.org/src/article/articleviewbase.cpp	2006-06-05 05:01:32.000000000 +0900
+++ jd-150_060601/src/article/articleviewbase.cpp	2006-06-05 04:56:52.000000000 +0900
@@ -869,7 +869,7 @@
         }
     }
 
-    for( int i = 1; i < m_current_bm ; ++i ){
+    for( int i = 1; i <= m_current_bm ; ++i ){
         if( m_article->is_bookmarked( i ) ){
             goto_num( i );
             m_current_bm = i;
diff -r -u jd-150_060601.org/src/article/drawareabase.cpp jd-150_060601/src/article/drawareabase.cpp
--- jd-150_060601.org/src/article/drawareabase.cpp	2006-05-22 23:30:11.000000000 +0900
+++ jd-150_060601/src/article/drawareabase.cpp	2006-06-05 04:56:52.000000000 +0900
@@ -1541,11 +1541,12 @@
     
     // search_move でひとつ進めるのでひとつ前に戻しておく
     if( ! reverse ){
-        ( *it ).select = false;
+        if( it != m_multi_selection.end() ) ( *it ).select = false;
         if( it == m_multi_selection.begin() )  m_multi_selection.back().select = true;
         else ( *( --it ) ).select = true;            
     }
     
+    redraw_view();
     return true;
 }
 
diff -r -u jd-150_060601.org/src/bbslist/bbslistviewbase.cpp jd-150_060601/src/bbslist/bbslistviewbase.cpp
--- jd-150_060601.org/src/bbslist/bbslistviewbase.cpp	2006-05-13 23:13:35.000000000 +0900
+++ jd-150_060601/src/bbslist/bbslistviewbase.cpp	2006-06-05 04:56:52.000000000 +0900
@@ -528,6 +528,10 @@
     // マウスジェスチャ
     SKELETON::View::get_control().MG_start( event );
 
+    // ダブルクリック
+    m_dblclick = false;
+    if( event->type == GDK_2BUTTON_PRESS ) m_dblclick = true; 
+
     CORE::core_set_command( "switch_bbslist" );
 
     return true;
@@ -550,6 +554,10 @@
 
     m_path_selected = m_treeview.get_path_under_xy( (int)event->x, (int)event->y );
 
+    // ダブルクリックの処理のため一時的にtypeを切替える
+    GdkEventType type_copy = event->type;
+    if( m_dblclick ) event->type = GDK_2BUTTON_PRESS;
+
     // 板を開く
     bool openboard = SKELETON::View::get_control().button_alloted( event, CONTROL::OpenBoardButton );
     bool openboardtab = SKELETON::View::get_control().button_alloted( event, CONTROL::OpenBoardTabButton );
@@ -561,6 +569,8 @@
         show_popupmenu( m_path_selected );
     }
 
+    event->type = type_copy;
+
     return true;
 }
 
diff -r -u jd-150_060601.org/src/bbslist/bbslistviewbase.h jd-150_060601/src/bbslist/bbslistviewbase.h
--- jd-150_060601.org/src/bbslist/bbslistviewbase.h	2006-05-13 22:05:56.000000000 +0900
+++ jd-150_060601/src/bbslist/bbslistviewbase.h	2006-06-05 04:56:52.000000000 +0900
@@ -35,6 +35,9 @@
         Gtk::ScrolledWindow m_scrwin;
         JDLIB::ConstPtr< Gtk::CellRendererText > m_ren_text;
 
+        // ダブルクリック状態
+        bool m_dblclick;
+
         // ポップアップメニュー用
         Gtk::TreeModel::Path m_path_selected;
 
diff -r -u jd-150_060601.org/src/board/boardview.cpp jd-150_060601/src/board/boardview.cpp
--- jd-150_060601.org/src/board/boardview.cpp	2006-05-22 23:30:11.000000000 +0900
+++ jd-150_060601/src/board/boardview.cpp	2006-06-05 04:56:52.000000000 +0900
@@ -1071,6 +1071,10 @@
     // マウスジェスチャ
     SKELETON::View::get_control().MG_start( event );
 
+    // ダブルクリック
+    m_dblclick = false;
+    if( event->type == GDK_2BUTTON_PRESS ) m_dblclick = true; 
+
     return true;
 }
 
@@ -1109,7 +1113,11 @@
                   << x << " " << y << " " << cell_x << " " << cell_y << std::endl;
 #endif
 
-        // 板を開く
+        // ダブルクリックの処理のため一時的にtypeを切替える
+        GdkEventType type_copy = event->type;
+        if( m_dblclick ) event->type = GDK_2BUTTON_PRESS;
+
+        // スレを開く
         bool openarticle = SKELETON::View::get_control().button_alloted( event, CONTROL::OpenArticleButton );
         bool openarticletab = SKELETON::View::get_control().button_alloted( event, CONTROL::OpenArticleTabButton );
         if( openarticle || openarticletab ) open_row( path, openarticletab );
@@ -1127,6 +1135,8 @@
             }
             if( popupmenu ) popupmenu->popup( 0, gtk_get_current_event_time() );
         }
+
+        event->type = type_copy;
     }
 
     return true;
diff -r -u jd-150_060601.org/src/board/boardview.h jd-150_060601/src/board/boardview.h
--- jd-150_060601.org/src/board/boardview.h	2006-05-03 01:03:19.000000000 +0900
+++ jd-150_060601/src/board/boardview.h	2006-06-05 04:56:52.000000000 +0900
@@ -27,6 +27,9 @@
         Glib::RefPtr< Gtk::ListStore > m_liststore;
         Gtk::ScrolledWindow m_scrwin;
 
+        // ダブルクリック状態
+        bool m_dblclick;
+
         // ソートで使う変数
         int m_previous_col;
         bool m_ascend;
diff -r -u jd-150_060601.org/src/jdlib/miscutil.cpp jd-150_060601/src/jdlib/miscutil.cpp
--- jd-150_060601.org/src/jdlib/miscutil.cpp	2006-06-05 05:01:32.000000000 +0900
+++ jd-150_060601/src/jdlib/miscutil.cpp	2006-06-05 04:56:52.000000000 +0900
@@ -16,9 +16,7 @@
 std::list< std::string > MISC::get_lines( const std::string& str, bool rm_space ){
         
     std::list< std::string > lines;
-	size_t i = 0, i2 = 0;
-	int r = 0;
-	
+    size_t i = 0, i2 = 0, r = 0;
     while ( ( i2 = str.find( "\n", i ) ) != std::string::npos ){
         r = 0;
         if( str[ i2 - 1 ] == '\r' ) r = 1;
