--- chasen.c.org	2004-01-14 14:05:43.743230968 +0900
+++ chasen.c	2004-01-14 14:07:47.659392848 +0900
@@ -53,6 +53,17 @@
 };
 
 
+static void init_opt_arg(void)
+{
+  int result;
+  char *opt[2];
+  
+  opt[0] = "chasen";
+  opt[1] = NULL;
+  result  = chasen_getopt_argv(opt, NULL);
+}
+
+
 void initchasen(void)
 {
   PyObject *m, *d;
@@ -66,4 +77,5 @@
   if (PyErr_Occurred()){
     Py_FatalError("can't initialize module chasen");
   }
+  init_opt_arg();
 }
