--- makefile.std        2004-04-28 10:50:37.000000000 +0900
+++ makefile.gentoo     2004-04-28 16:15:10.000000000 +0900
@@ -63,2 +63,3 @@
 IRC=TRUE
+IRCARGS = -DIRC_SERVER=\"irc.kyoto.wide.ad.jp\" -DIRC_PORT=\"6667\" -DIRC_CHANNEL=\"\#angband\"
 
@@ -154,5 +155,5 @@
 #
-COPTS = -Wall -O1 -pipe -g
-INCLUDES = -I/usr/X11R6/include
-DEFINES = -DUSE_X11 \
+COPTS = -Wall -O1 -pipe
+INCLUDES = -I/usr/X11R6/include -I/usr/include/ncurses
+DEFINES = -DUSE_GCU -DUSE_X11 -DUSE_NCURSES \
        -DUSE_EGO_GRAPHICS -DUSE_TRANSPARENCY -DSUPPORT_GAMMA \
@@ -160,3 +161,3 @@
        -DUSE_PRECISE_CMOVIE -DUSE_UNIXSOCK
-LIBS = -lX11 -L/usr/X11R6/lib
+LIBS = -lX11 -L/usr/X11R6/lib -lncurses

