diff -Naur perlftlib-1.2/FreeType/FreeType.xs perlftlib-1.2-gentoo/FreeType/FreeType.xs
--- perlftlib-1.2/FreeType/FreeType.xs	1998-12-14 14:13:26.000000000 +0900
+++ perlftlib-1.2-gentoo/FreeType/FreeType.xs	2002-09-09 01:58:01.000000000 +0900
@@ -16,7 +16,7 @@
 }
 #endif
 
-
+static int na;
 #define STORE_OBJ(obj, elem, sv)\
 	hv_store((obj), #elem, sizeof(#elem)-1, sv, 0)
 #define STORE_OBJ_IV(obj, var, elem)\
diff -Naur perlftlib-1.2/FreeType/Makefile.PL perlftlib-1.2-gentoo/FreeType/Makefile.PL
--- perlftlib-1.2/FreeType/Makefile.PL	1998-12-14 14:13:26.000000000 +0900
+++ perlftlib-1.2-gentoo/FreeType/Makefile.PL	2002-09-09 01:58:01.000000000 +0900
@@ -4,7 +4,7 @@
 WriteMakefile(
     'NAME'	=> 'FreeType',
     'VERSION_FROM' => 'FreeType.pm', # finds $VERSION
-    'LIBS'	=> ['-L/usr/local/X11R6.3/lib -lttf'],   # e.g., '-lm' 
+    'LIBS'	=> ['-L/usr/X11R6/lib -lttf'],   # e.g., '-lm' 
     'DEFINE'	=> '',     # e.g., '-DHAVE_SOMETHING' 
-    'INC'	=> '-I/usr/local/X11R6.3/include',     # e.g., '-I/usr/include/other' 
+    'INC'	=> '-I/usr/include/freetype',     # e.g., '-I/usr/include/other' 
 );
diff -Naur perlftlib-1.2/Makefile perlftlib-1.2-gentoo/Makefile
--- perlftlib-1.2/Makefile	1998-12-14 14:13:26.000000000 +0900
+++ perlftlib-1.2-gentoo/Makefile	2002-09-09 01:58:01.000000000 +0900
@@ -1,8 +1,8 @@
-
-BINDIR = /usr/local/bin
-MANDIR = /usr/local/man/man1
+PREFIX = /usr
+BINDIR = $(PREFIX)/bin
+MANDIR = $(PREFIX)/man/man1
 PERL = perl
-PERLLIBDIR = /usr/local/lib/perl5/site_perl
+PERLLIBDIR = $(PREFIX)/lib/perl5/site_perl
 
 VERSION = 1.2
 
@@ -13,10 +13,10 @@
 
 install: installlib $(PROGRAMS)
 	cp $(PROGRAMS) $(BINDIR)
-	rm -f $(MANDIR)/ftinfo.1
-	ln -s $(BINDIR)/ftinfo $(MANDIR)/ftinfo.1
-	rm -f $(MANDIR)/mkttfdir.1
-	ln -s $(BINDIR)/mkttfdir $(MANDIR)/mkttfdir.1
+#	rm -f $(MANDIR)/ftinfo.1
+#	ln -s $(BINDIR)/ftinfo $(MANDIR)/ftinfo.1
+#	rm -f $(MANDIR)/mkttfdir.1
+#	ln -s $(BINDIR)/mkttfdir $(MANDIR)/mkttfdir.1
 
 clean realclean distclean: FORCE
 	cd FreeType; $(MAKE) $(MFLAGS) $@
diff -Naur perlftlib-1.2/ftinfo.pl perlftlib-1.2-gentoo/ftinfo.pl
--- perlftlib-1.2/ftinfo.pl	1998-12-14 14:13:26.000000000 +0900
+++ perlftlib-1.2-gentoo/ftinfo.pl	2002-09-09 01:58:33.000000000 +0900
@@ -1,4 +1,4 @@
-#! /usr/local/bin/perl
+#! /usr/bin/perl
 'di ';
 'ds 00 \"';
 'ig 00 ';
diff -Naur perlftlib-1.2/mkttfdir.pl perlftlib-1.2-gentoo/mkttfdir.pl
--- perlftlib-1.2/mkttfdir.pl	1998-12-14 14:13:26.000000000 +0900
+++ perlftlib-1.2-gentoo/mkttfdir.pl	2002-09-09 01:58:23.000000000 +0900
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 'di ';
 'ds 00 \"';
 'ig 00 ';
