--- R/src/modules/X11/dataentry.c.orig	Fri Jan 30 05:08:22 2004
+++ R/src/modules/X11/dataentry.c	Fri Jan 30 05:05:11 2004
@@ -26,11 +26,12 @@
 #include <stdlib.h>

 /* don't use X11 function prototypes (which tend to ...): */
-#define NeedFunctionPrototypes 0
+#define NeedFunctionPrototypes 1
 /* XFree 4.4.0 forgets to define this if NeedFunctionPrototypes=0 */
 #ifndef _Xconst
 #define _Xconst const
 #endif
+#include <X11/Xfuncproto.h>
 #include <X11/X.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
