--- cannasocket.c.orig	2004-01-05 00:48:51.000000000 +0900
+++ cannasocket.c	2004-01-05 00:50:14.000000000 +0900
@@ -197,24 +197,16 @@
       localhost_inaddr = 0;
     }
 
-    if (addr_in.sin_addr.s_addr != myhost_inaddr &&
-	addr_in.sin_addr.s_addr != localhost_inaddr) {
-      m_netaddr2ascii(addr_in.sin_addr.s_addr, buf);
-
-      m_msg("REFUSE THE CONNECTION REQUEST FROM %s.\n", buf);
-      close(newfd);
-      return -1;
-    } else {
-      m_netaddr2ascii(addr_in.sin_addr.s_addr, buf);
+    m_netaddr2ascii(addr_in.sin_addr.s_addr, buf);
 
-      hp = gethostbyaddr((char *)&addr_in.sin_addr, sizeof(struct in_addr),
+    hp = gethostbyaddr((char *)&addr_in.sin_addr, sizeof(struct in_addr),
 			 AF_INET);
 
-      if (hp && hp->h_name)
+    if (hp && hp->h_name)
 	host = strdup(hp->h_name);
-      else
+    else
 	host = strdup(buf);
-    }
+
   } else {
     host = strdup("UNIX");
   }
