*** catclock.orig/imakefile	2005-03-01 15:52:32.709851054 +0900
--- catclock/imakefile	2005-03-01 15:35:49.636285313 +0900
***************
*** 9,15 ****
     DEFINES = -DHAS_GNU_EMACS
  
  LOCAL_LIBRARIES = $(MOTIF_LIBS)
!   SYS_LIBRARIES = -lX11 -lm
  
  
  all:: catclock
--- 9,15 ----
     DEFINES = -DHAS_GNU_EMACS
  
  LOCAL_LIBRARIES = $(MOTIF_LIBS)
!   SYS_LIBRARIES = -lX11 -lm -lXm
  
  
  all:: catclock
