--- Makefile.orig	2000-11-29 13:29:59.000000000 +0900
+++ Makefile	2004-01-09 06:57:19.363685448 +0900
@@ -55,50 +55,50 @@
 	@${SH} ./WXGINSTALL.sh $(OSTYPE) PRE-INSTALL
 
 wxg		:
-	@install -d $(DICDIR)
-	@install -d $(RCDIR)
+	@install -d $(DESTDIR)$(DICDIR)
+	@install -d $(DESTDIR)$(RCDIR)
 	@tar xfz $(WXG)
-	@install -m 4755 wxg/wxgserver $(SBINDIR)
-	@install wxg/wxgdic $(BINDIR)
-	@install wxg/wxgdmp $(BINDIR)
-	@install wxg/wxgdset $(BINDIR)
-	@install wxg/$(RCFILE) $(RCDIR)
-	@install -m 0644 wxg/wxgrc $(WXGDIR)
+	@install -m 4755 wxg/wxgserver $(DESTDIR)$(SBINDIR)
+	@install wxg/wxgdic $(DESTDIR)$(BINDIR)
+	@install wxg/wxgdmp $(DESTDIR)$(BINDIR)
+	@install wxg/wxgdset $(DESTDIR)$(BINDIR)
+	@install wxg/$(RCFILE) $(DESTDIR)$(RCDIR)
+	@install -m 0644 wxg/wxgrc $(DESTDIR)$(WXGDIR)
 	@$(RM) wxg/*
 	@$(RMDIR) wxg
 
 wxgdics		:
-	@install -d $(WXGDIR)
-	@install -d $(ROMADIR)
-	@install -d $(DICDIR)
+	@install -d $(DESTDIR)$(WXGDIR)
+	@install -d $(DESTDIR)$(ROMADIR)
+	@install -d $(DESTDIR)$(DICDIR)
 	@tar xfz wxgdics.tar.gz
-	@install -m 0644 dic/wxg.gct $(WXGDIR)
+	@install -m 0644 dic/wxg.gct $(DESTDIR)$(WXGDIR)
 	@$(RM) dic/wxg.gct
-	@install -m 0644 dic/wxgroma.grt $(ROMADIR)
+	@install -m 0644 dic/wxgroma.grt $(DESTDIR)$(ROMADIR)
 	@$(RM) dic/wxgroma.grt
-	@$(BINDIR)/wxgdset dic/wxg.gds -L=0 -U=2 \
+	@$(DESTDIR)$(BINDIR)/wxgdset dic/wxg.gds -L=0 -U=2 \
 		wxgu.gdj -p1 wxg.gdj -p1 wxgtx.gdj -p1 seimei.gdj -p1 chimei.gdj -p1 \
 		yubin.gdj -p1 kao.gdj -p1 english.gdj -p1
-	@$(BINDIR)/wxgdset dic/wxgsei.gds -L=0 -U=2 wxgu.gdj -p1 wxg.gdj -p1 seimei.gdj \
+	@$(DESTDIR)$(BINDIR)/wxgdset dic/wxgsei.gds -L=0 -U=2 wxgu.gdj -p1 wxg.gdj -p1 seimei.gdj \
 		-p1 chimei.gdj -p1 yubin.gdj -p1 tel.gdj -p1
 
-	@$(BINDIR)/wxgdset dic/wxgu.gds -C=$(DICABSDIR)/ -L=0 -U=2 wxgu.gdj -p1 wxg.gdj -p3 \
+	@$(DESTDIR)$(BINDIR)/wxgdset dic/wxgu.gds -C=$(DICABSDIR)/ -L=0 -U=2 wxgu.gdj -p1 wxg.gdj -p3 \
 		wxgtx.gdj -p3 seimei.gdj -p3 chimei.gdj -p3 yubin.gdj -p3 kao.gdj -p3 \
 		english.gdj -p3
 
-	@$(BINDIR)/wxgdset dic/wxgseiu.gds -C=$(DICABSDIR)/ -L=0 -U=2 wxgu.gdj -p1 wxg.gdj -p3 \
+	@$(DESTDIR)$(BINDIR)/wxgdset dic/wxgseiu.gds -C=$(DICABSDIR)/ -L=0 -U=2 wxgu.gdj -p1 wxg.gdj -p3 \
 		seimei.gdj -p3 chimei.gdj -p3 yubin.gdj -p3 tel.gdj -p3
 
-	@install -m 0644 dic/* $(DICDIR)
+	@install -m 0644 dic/* $(DESTDIR)$(DICDIR)
 	@$(RM) dic/*
 	@$(RMDIR) dic
 
 wxgdoc		:
-	@install -d $(DOCDIR)
-	@$(CP) README $(DOCDIR)
-	@install -d $(HTMLDIR)
+	@install -d $(DESTDIR)$(DOCDIR)
+	@$(CP) README $(DESTDIR)$(DOCDIR)
+	@install -d $(DESTDIR)$(HTMLDIR)
 	@tar xfz wxghtml.tar.gz
-	@install -m 0644 html/* $(HTMLDIR)
+	@install -m 0644 html/* $(DESTDIR)$(HTMLDIR)
 	@$(RM) html/*
 	@$(RMDIR) html
 
