diff -ur sjxa-1.6.14/resource/Imakefile sjxa-1.6.14-gentoo/resource/Imakefile
--- sjxa-1.6.14/resource/Imakefile	Wed Mar 29 15:33:35 1995
+++ sjxa-1.6.14-gentoo/resource/Imakefile	Tue Jan  3 11:01:15 2006
@@ -57,6 +57,6 @@
 #if HasSjisLocale
 InstallLocaleAppDefaults(Sjxa,sjis,SJXAAPPDIR,ja_JP.SJIS)
 #endif
-InstallLocaleAppDefaults(Sjxa,eucJP,SJXAAPPDIR,ja_JP.EUC)
+InstallLocaleAppDefaults(Sjxa,eucJP,SJXAAPPDIR,ja_JP.eucJP)
 
 InstallNamedNonExec(sjxark,sjxark,SJXARKDIR)
diff -ur sjxa-1.6.14/sjxa.c sjxa-1.6.14-gentoo/sjxa.c
--- sjxa-1.6.14/sjxa.c	Tue Sep 15 15:34:26 1998
+++ sjxa-1.6.14-gentoo/sjxa.c	Tue Jan  3 10:59:49 2006
@@ -146,12 +146,14 @@
 
    /* get user locale */
     if( (locale_name = setlocale( LC_CTYPE, "" )) == NULL ) {
-	fprintf( stderr, "%s error: locale could not be set.\n", program_name );
-	exit( 1 );
+	if( (locale_name = getenv("LANG")) == NULL ) {
+	    fprintf( stderr, "%s error: locale could not be set.\n", program_name );
+	    exit( 1 );
+	}
     }
     if( strcmp( locale_name, "ja_JP.SJIS" ) == 0 )
 	user_locale = LC_CTYPE_SHIFTJIS;
-    else if( strncmp( locale_name, "ja_JP.EUC", 2 ) == 0 )
+    else if( strncmp( locale_name, "ja_JP.eucJP", 2 ) == 0 )
 	user_locale = LC_CTYPE_EUC;
     else if( strstr( locale_name, "ja" ) != NULL )
 	user_locale = LC_CTYPE_EUC;
