diff -ur sjxa-1.6.14/CONFIG sjxa-1.6.14-gentoo/CONFIG
--- sjxa-1.6.14/CONFIG	Tue Sep 15 15:34:21 1998
+++ sjxa-1.6.14-gentoo/CONFIG	Tue Jan  3 12:11:54 2006
@@ -112,19 +112,19 @@
 
 #else /* default define */			/**/
 
-#define UseInstalledSj3Lib	NO
+#define UseInstalledSj3Lib	YES
 /*#define Sj3LibDir		$(TOP)/usr/lib*/
 #define HasSjisLocale		NO
 #define HasJlsFunction		NO
 #define UseXIMCP		YES
-#define UseXIMP			YES
-#define UseXLC			YES
+#define UseXIMP		NO
+#define UseXLC			YES
 #define UseSJX			NO
-/*#define JapaneseManLocale	eucJP*/
-/*#define JapaneseManDir		/usr/local/man/ja_JP.EUC/man1*/
-/*#define ManDir			/usr/local/man/C/man1*/
-/*#define SjxaBinDir		/usr/local/bin/X11*/
-/*#define SjxaAppDir		/usr/local/lib/X11*/
-/*#define SjxarkDir		/usr/local/lib/X11/sjxa*/
+#define JapaneseManLocale	eucJP
+#define JapaneseManDir		/usr/share/man/ja/man1
+#define ManDir			/usr/share/man/man1
+#define SjxaBinDir		/usr/bin/X11
+#define SjxaAppDir		/usr/lib/X11
+#define SjxarkDir		/usr/lib/X11/sjxa
 
 #endif
diff -ur sjxa-1.6.14/Xmw/View.c sjxa-1.6.14-gentoo/Xmw/View.c
--- sjxa-1.6.14/Xmw/View.c	Mon Nov 28 14:09:41 1994
+++ sjxa-1.6.14-gentoo/Xmw/View.c	Tue Jan  3 10:20:22 2006
@@ -381,7 +381,7 @@
 	    }
 	    GetGeometry( widget, child->core.width, child->core.height );
 	    (*((ViewportWidgetClass)w->core.widget_class)->form_class.layout)
-		( (FormWidget)w, w->core.width, w->core.height );
+		( (FormWidget)w, w->core.width, w->core.height, True );
 	    /* %%% do we need to hide this child from Form?  */
 	}
     }
diff -ur sjxa-1.6.14/sj3libP.h sjxa-1.6.14-gentoo/sj3libP.h
--- sjxa-1.6.14/sj3libP.h	Tue Sep 15 15:34:30 1998
+++ sjxa-1.6.14-gentoo/sj3libP.h	Sun Jan  1 21:15:20 2006
@@ -48,7 +48,7 @@
 #define ProtoName		"tcp"
 #define	PortNumber		"3000"
 #else
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(linux)
 #define	PortNumber		3086
 #else
 #define	PortNumber		3000
