--- /var/tmp/portage/fontconfig-2.2.1-r1/work/fontconfig-2.2.1/src/fcfreetype.c	2005-04-01 16:35:09.673881728 +0900
+++ fcfreetype.c	2005-04-01 16:36:35.521830856 +0900
@@ -57,7 +57,6 @@
 
 #if (FREETYPE_MINOR > 1 || (FREETYPE_MINOR == 1 && FREETYPE_PATCH >= 4))
 #include <freetype/ftbdf.h>
-#include <freetype/ftmodule.h>
 #define USE_FTBDF
 #define HAS_BDF_PROPERTY(f) ((f) && (f)->driver && \
 			     (f)->driver->root.clazz->get_interface)
